uwblueprint/extend-a-family
# added discriminator for learner and facilitator

## Notion ticket link
[Facilitator & Learner Relationship](https://www.notion.so/uwblueprintexecs/Facilitator-Learner-Relationship-20a31631fae246169c51ecb36c44b35c)



## Implementation description
* Added to base user model for learner and facilitator discriminators
* Added learner property to facilitators (list of strings)
* Added Facilitator property to learners (singular string)
* Created test in server.ts for saving to these user types



## Steps to test
1. Set up the .env with MG_DATABASE_NAME set to <name>_dev.
2. Run the server and verify that you can create and save documents with the new user types.
3. Check that the learners property for facilitators defaults to an empty array if not provided.
4. Ensure the facilitator property for learners is required and correctly assigned.




## What should reviewers focus on?
* Ensure the discriminators for the User model are correctly implemented.
* Verify the default value for the learners property.
* Confirm the facilitator property is required for learners and functions as expected.


## Checklist
